src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
+                                }
+                                else
+                                {
+                                    skip_feature = true;
+                                }
+                            }
+                            else // safeguard – should never reach here
+                            {
+                                skip_feature = true;
+                            }
                         }
 
-                        if (property_key == baton_data_->class_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->class_property) ||
+                            utils::startswith(property_key, baton_data_->class_prefix + baton_data_->class_property))
                         {
-                            if (property.value().type() == vtzero::property_value_type::string_value)
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(class_key_precedence.begin(), class_key_precedence.end(), property_key) - class_key_precedence.begin());
+                            if (idx < class_key_idx)
                             {
+                                class_key_idx = idx;
                                 class_value = property.value();
                             }
-                            continue;
+                            // wait till we are done looping through all properties before we add class value to final_properties
                         }
 
-                        // preserve original params.language_property value
-                        if (property_key == baton_data_->language_property)
+                        else if (
+                            utils::startswith(property_key, baton_data_->language_property) ||
+                            utils::startswith(property_key, baton_data_->language_prefix + baton_data_->language_property))
                         {
-                            name_value = property.value();
+                            // check if the property is of higher precedence that class key encountered so far
+                            std::uint32_t idx = static_cast<std::uint32_t>(std::find(language_key_precedence.begin(), language_key_precedence.end(), property_key) - language_key_precedence.begin());
+                            if (idx < language_key_idx)
+                            {
+                                language_key_idx = idx;
+                                language_value = property.value();
+                            }
 
-                            // if no language was specified, we want the name value to be constant
-                            if (baton_data_->language.empty())
+                            // preserve original language value and wait till finish looping through all properties to assign a value
```

Comment:
for my understanding, won't there be a single property with this key? why wait before assigning a value?

Reply:
`original_language_value` is only used to assign `{language_property}_local` (if the feature has no available language, `{language_property}` will also have the same value but that'll come from the variable `language_value`). I postpone the assigning of `{language_property}_local` till the end because I think that makes the code easier to read for humans:
- everything that's assigned inside the properties for-loop are properties we are not modifying;
- everything assigned after the for-loop are properties we change/add because of localization.